runtime/manta/src/migrations/assets_genesis.rs

```diff
@@ -0,0 +1,397 @@
+};
+use sp_runtime::BoundedVec;
+use sp_std::vec::Vec;
+
+pub type DepositBalanceOf<T, I = ()> = <<T as pallet_assets::Config<I>>::Currency as Currency<
+    <T as frame_system::Config>::AccountId,
+>>::Balance;
+
+pub const INITIAL_PALLET_ASSETS_MANAGER_VERSION: u16 = 0;
+pub const INITIAL_PALLET_ASSETS_VERSION: u16 = 0;
+
+pub struct AssetsGenesis<T>(PhantomData<T>);
+impl<T> OnRuntimeUpgrade for AssetsGenesis<T>
+where
+    T: pallet_asset_manager::Config + pallet_assets::Config,
+{
+    fn on_runtime_upgrade() -> Weight {
+        let asset_manager_storage_version =
+            <pallet_asset_manager::Pallet<T> as GetStorageVersion>::on_chain_storage_version();
+        let assets_storage_version =
+            <pallet_assets::Pallet<T> as GetStorageVersion>::on_chain_storage_version();
+        if asset_manager_storage_version != INITIAL_PALLET_ASSETS_MANAGER_VERSION
+            || assets_storage_version != INITIAL_PALLET_ASSETS_VERSION
+        {
+            log::info!("Aborting migration due to unexpected on-chain storage versions for pallet-assets-manager: {:?} and pallet-assets: {:?}. Expectation was: {:?} and {:?}.", asset_manager_storage_version, assets_storage_version, INITIAL_PALLET_ASSETS_MANAGER_VERSION, INITIAL_PALLET_ASSETS_VERSION );
+            return T::DbWeight::get().reads(2);
+        }
+
+        log::info!(target: "asset-manager", "Starting migration for AssetManager...");
+
+        let pallet_prefix: &[u8] = b"AssetManager";
+        let _ = clear_storage_prefix(pallet_prefix, b"AssetIdLocation", b"", None, None);
+        let _ = clear_storage_prefix(pallet_prefix, b"LocationAssetId", b"", None, None);
+        let _ = clear_storage_prefix(pallet_prefix, b"AssetIdMetadata", b"", None, None);
+        let _ = clear_storage_prefix(pallet_prefix, b"UnitsPerSecond", b"", None, None);
+        let _ = clear_storage_prefix(pallet_prefix, b"MinXcmFee", b"", None, None);
+        let _ = clear_storage_prefix(pallet_prefix, b"AllowedDestParaIds", b"", None, None);
+
+        put_storage_value(
+            pallet_prefix,
+            b"NextAssetId",
+            b"",
+            <T::AssetConfig as AssetConfig<T>>::StartNonNativeAssetId::get(),
+        );
+        let asset_id = <T::AssetConfig as AssetConfig<T>>::NativeAssetId::get();
+        let metadata = <T::AssetConfig as AssetConfig<T>>::NativeAssetMetadata::get();
+        let location = <T::AssetConfig as AssetConfig<T>>::NativeAssetLocation::get();
+        put_storage_value(
+            pallet_prefix,
+            b"AssetIdLocation",
+            &Blake2_128Concat::hash(&asset_id.encode()),
+            location.clone(),
+        );
+        put_storage_value(
+            pallet_prefix,
+            b"LocationAssetId",
+            &Blake2_128Concat::hash(&location.encode()),
+            asset_id,
+        );
+        put_storage_value(
+            pallet_prefix,
+            b"AssetIdMetadata",
+            &Blake2_128Concat::hash(&asset_id.encode()),
```

Comment:
`setAssetIdMetadata` has a component that writes some metadata in `pallet-assets`. You're not writing anything in `pallet-assets` in the migration later, i think this will leave this metadata incomplete

Reply:
hm, asset manager genesis build does the same thing
https://github.com/Manta-Network/Manta/blob/dc29283c438d99cf9b1cefe23d9e9ccf0e9c7617/pallets/asset-manager/src/lib.rs#L184

Reply:
the part that will be missing for the native asset is https://github.com/Manta-Network/Manta/blob/dc29283c438d99cf9b1cefe23d9e9ccf0e9c7617/runtime/calamari/src/assets_config.rs#L121

Reply:
idk if that means it has always been buggy or it's safe for some reason

Reply:
Correct.
This is actually equivalent in Calamari. The Assets pallet doesn't know anything about the native asset.
This has added benefit of the MANTA asset not showing up in Assets page in the front-end (with 0 issuance)

Reply:
What i'm hoping to get here is assurance that this lack-of-asset-metadata-on-an-otherwise-normally-registered-asset can not lead to some sort of runtime crash somewhere down the road.

If you can with certainty state this won't happen i'll trust in that, but if not we should have a meeting next week to make sure everyone fully understands the implications of this. whether this is already deployed or not

Reply:
I mean i haven't done formal verification or something, but there are no panics in the assets code, so we should be safe from random crashes.
Only the asset manager and xcm are using the assets pallet, and they both won't have an issue handling missing asests during lookups. For example you remember TooExpensive xcm errors.

Reply:
And if we do add it the total balance will always be 0 (wrong) , and only valid part would be the metadata.